apps/aehttp/test/aehttp_hyperchains_SUITE.erl

```diff
@@ -718,6 +727,79 @@ epoch_with_slow_parent(Config) ->
                  [rpc(?NODE1, aec_chain_hc, epoch_start_height, [N]) || N <- lists:seq(1, EndEpoch)]),
     ok.
 
+%%%=============================================================================
+%%% Pinning
+%%%=============================================================================
+
+get_pin(Config) ->
+
+    %% note: we are actuall in epoch=2, but the pins are for the previous epoch
+    Repl1 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 1}} = Repl1,
+    %% product, still in same epoch, so pins should be the same
+    {ok, _} = produce_cc_blocks(Config, 2),
+    Repl2 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 1}} = Repl2,
+    ?assertEqual(Repl1, Repl2),
+    %% move into next epoch, test pin:epoch
+    {ok, _} = produce_cc_blocks(Config, 9),
+    Repl3 = aecore_suite_utils:http_request(aecore_suite_utils:external_address(), get, "hyperchain/pin-tx", []),
+    {ok, 200, #{<<"epoch">> := 2}} = Repl3,
+
+    ok.
+
+post_pin_to_pc(Config) ->
+
+    %% we use local/rpc here.
+    Pin = rpc(?NODE1, aec_pinning_agent, get_pinning_data, []),
+    PinPayloadBin = rpc(?NODE1, aec_pinning_agent, encode_pin_payload, [Pin]),
```

Comment:
Should that not be compared to the data you got from the http request in previous test... in other words, make it into one test and compare

Reply:
Let's improve in next PR...